### PR TITLE
CMake pure build on Windows

### PR DESCRIPTION
**Pull request by XeCycle**
Building on Windows has long been known to fail (#48), but if we skip the TCP part and do a pure build, it is easy to fix.  This adds a `-DPURE_BUILD=ON` option to CMake build, and fixes it on Windows, tested with VS 2015.  Personally I use the lib integrated into libuv, these should be enough for me to continue hacking.

Btw `tcpdefines.h` added by #80 was moved to `src/`; to me it looks like a simple internal header.  No idea whether you would like to keep it there (and have it installed).

### CODE REVIEW

File: src/includes.h [hunk 1]
```diff
@@ -22,10 +22,14 @@
 #include <vector>
 #include <queue>
 #include <sys/types.h>
+
+#ifndef _WIN32
```
Comment (ricejasonf) [inline src/includes.h]: Why not skip these for "BUILD_PURE" instead of just WIN32?

File: CMakeLists.txt [hunk 2]
```diff
@@ -48,6 +48,8 @@ if(BUILD_TUTORIALS)
 #    add_subdirectory(examples/rabbitmq_tutorials)
 endif()
 
+option(BUILD_PURE "build in pure mode" OFF)
```
Comment (ricejasonf) [inline CMakeLists.txt]: It would be a good idea to prefix `BUILD_PURE` with something unique. Like `AMQP_CPP_BUILD_PURE`.

Comment (XeCycle) [inline CMakeLists.txt]: I've seen projects following either way, so maybe it is a bit controversial here.  Some may feel it like `--with-tutorials` or `--without-tcp`, and others still are afraid of name conflicts.

Name conflicts come into play when this project is used by `add_subdirectory` from another project.  Looking at that set to parent scope, seems we are to support this use case?

Personally I prefer to prefix it, but here I'm just following existing code --- several lines above, `BUILD_TUTORIALS`.

File: src/includes.h (hunk 1 above)
Comment (XeCycle) [inline src/includes.h]: Good catch, will include that.

File: CMakeLists.txt [hunk 3]
```diff
@@ -49,6 +49,9 @@ if(BUILD_TUTORIALS)
 endif()
 
 option(BUILD_PURE "build in pure mode" OFF)
+if(BUILD_PURE)
+    add_definitions(AMQP_CPP_BUILD_PURE)
```
Comment (tretiy) [inline CMakeLists.txt]: Seems that AMQP_CPP_BUILD_PURE should have -D prefix

**Comment by tommitytom** [comment]
What's the go with this?  Has it been merged in to the main repo?

**Comment by LiamKarlMitchell** [comment]
So can we build on Windows VS2017?

**Comment by surgura** [comment]
This has been fixed/sortof merged as of 91c4f891590c630df4f2a18e3f5925ddf41b3d3f, so we can close.

**Pull request closed without merging**